rust-connectors/sinks/kafka/src/main.rs

```diff
@@ -18,15 +26,55 @@ async fn main() -> anyhow::Result<()> {
         println!("{}", serde_json::to_string_pretty(&schema).unwrap());
         return Ok(());
     }
-    let opts: KafkaOpt = KafkaOpt::from_args();
-    opts.common.enable_logging();
+    let raw_opts = KafkaOpt::from_args();
+    raw_opts.common.enable_logging();
     info!(
         connector_version = env!("CARGO_PKG_VERSION"),
         git_hash = git_hash_version(),
         "Starting Kafka sink connector",
     );
 
-    opts.execute().await?;
+    // =======================================================
+    // Setup objects and actors based on opts
+    // =======================================================
+
+    // KafkaSinkWorker members need to be inside Arc because it is referenced inside tokio::spawn
+    // That expect things moved inside to be static
+    // An optimization point for this would be to own the runtime (rather than calling tokio::*)
+    // So that runtime::spawn does not need producer to be in Arc
+    let kafka_sink_deps =
+        KafkaSinkDependencies::from_kafka_opt(raw_opts).await? as KafkaSinkDependencies;
```

Comment:
Oh this one is a useless artifact. My bad. I'm removing it.

Previously it was FutureProducer that requires `as`, maybe because of ambiguous type inference on rustc's part.